base/string_utils.cpp

```diff
@@ -283,6 +283,18 @@ std::u16string ToUtf16(std::string_view utf8)
   return utf16;
 }
 
+UniString Substring(UniString const & str, size_t start, size_t length)
+{
+  // Ensure start is within bounds
+  if (start >= str.size())
```

Comment:
Let's make it a `CHECK_LESS(start, str.size(), ());` to immediately crash and fix the bug.